### PR TITLE
Add a feature to switch back to the legacy Swift driver in Swift 5.5 and higher.

### PR DESCRIPTION
**Pull request by brentleyjones**
PiperOrigin-RevId: 398758812
(cherry picked from commit 69bc3ede9176a02207b916a3c87801ce30639fb2)

### CODE REVIEW

**Comment by brentleyjones** [comment]
(I'll remember to Rebase and Merge with this one...)

File: swift/internal/compiling.bzl [hunk 1]
```diff
@@ -129,8 +130,23 @@ def compile_action_configs(
...
     """
 
-    #### Flags that control compilation outputs
+    #### Flags that control the driver
     action_configs = [
+        # Use the legacy driver if requested.
+        swift_toolchain_config.action_config(
+            actions = [
+                swift_action_names.COMPILE,
+                swift_action_names.PRECOMPILE_C_MODULE,
```
Comment (keith) [inline swift/internal/compiling.bzl:139]: probably needs to apply to the split module action as well?

Comment (brentleyjones) [reply swift/internal/compiling.bzl:139]: Is it needed for `swift_action_names.DUMP_AST`?

Comment (keith) [reply swift/internal/compiling.bzl:140]: Probably any action invoking swift.

**Review by thii — APPROVED** [review]

**Pull request merged**